torch/csrc/jit/codegen/cuda/scheduler/pointwise.cpp

```diff
@@ -822,37 +821,27 @@ void schedulePointwise(Fusion* fusion, const PointwiseParams& params) {
   TransformPropagator propagator(reference_tv);
   MaxRootDomainInfoSpanningTree spanning_tree(reference_tv);
   spanning_tree.traverse(&propagator);
-  scheduler_utils::parallelizeAllLike(reference_tv, all_tvs);
+  scheduler_utils::parallelizeAllLike(reference_tv);
 
   if (params.vectorize) {
     // Grab all tensor views that should be vectorized
-    auto vectorized_tvs =
+    auto inputs_outputs =
         scheduler_utils::getInputsOutputsWithInnerDim(reference_tv, true);
-    // Going to move inputs to consumers of inputs, need a copy as we'll modify
-    // the original.
-    {
-      auto vectorized_tvs_copy = vectorized_tvs;
-      for (auto inp : vectorized_tvs_copy) {
-        if (!inp->isFusionInput()) {
-          continue;
-        }
-        vectorized_tvs.erase(
-            std::find(vectorized_tvs.begin(), vectorized_tvs.end(), inp));
-        auto consumer_tvs = ir_utils::consumerTvsOf(inp);
-        vectorized_tvs.insert(
-            vectorized_tvs.end(), consumer_tvs.begin(), consumer_tvs.end());
+    std::unordered_set<TensorView*> vectorized_tvs;
+    for (auto tv : inputs_outputs) {
+      if (!tv->isFusionInput()) {
+        vectorized_tvs.emplace(tv);
+        continue;
       }
+      // move inputs to consumers of inputs
+      auto consumer_tvs = ir_utils::consumerTvsOf(tv);
+      vectorized_tvs.insert(consumer_tvs.begin(), consumer_tvs.end());
     }
-    // Clear vectorize on tensors that shouldn't have it
-    for (auto tv : all_tvs) {
-      if (std::find(vectorized_tvs.begin(), vectorized_tvs.end(), tv) ==
-          vectorized_tvs.end()) {
-        for (auto id : tv->domain()->domain()) {
-          if (id->getParallelType() == ParallelType::Vectorize) {
-            id->parallelize(ParallelType::Serial);
-          }
-        }
-      }
+    vectorize_id->parallelize(ParallelType::Vectorize);
+    scheduler_utils::parallelizeAllLike(
+        reference_tv, vectorized_tvs, {ParallelType::Vectorize});
+    if (vectorized_tvs.count(reference_tv) == 0) {
+      vectorize_id->parallelize(ParallelType::Serial);
```

Comment:
This is just a cleanup of the code using the new functionality. There should be no behavioral change.